sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -376,11 +372,6 @@ public void finalizeCheckpoint() throws IOException {
       this.source = source;
       this.current = null;
       this.checkpointMark = checkpointMark != null ? checkpointMark : new RabbitMQCheckpointMark();
-      try {
```

Comment:
I've addressed this in another PR https://github.com/apache/beam/pull/9820/files#r335994291 but tl;dr: the ConnectionHandler used in the constructor and `close` is not the same one used in `start`. I now defer initializing the class-level ConnectionHandler to `start` as it feels more in-line with start/stop semantics.